// grunt-angular-protractor/tests/task2/conf.js
// An example configuration file.
exports.config = {
    directConnect: false,

    // Capabilities to be passed to the webdriver instance.
    multiCapabilities: [
        {

            browserName: 'chrome',

            // Name of the process executing this capability.  Not used directly by
            // protractor or the browser, but instead pass directly to third parties
            // like SauceLabs as the name of the job running this test
            name: 'chrome Job',

            // User defined name for the capability that will display in the results log
            // Defaults to the browser name
            logName: 'Chrome',

            // Number of times to run this set of capabilities (in parallel, unless
            // limited by maxSessions). Default is 1.
            count: 1,

            // If this is set to be true, specs will be sharded by file (i.e. all
            // files to be run by this set of capabilities will run in parallel).
            // Default is false.
            shardTestFiles: true,

            // Maximum number of browser instances that can run in parallel for this
            // set of capabilities. This is only needed if shardTestFiles is true.
            // Default is 1.
            maxInstances: 1,

            // Additional spec files to be run on this capability only.
            specs: ['*_e2e.js']

        },
        {

            browserName: 'firefox',

            // Name of the process executing this capability.  Not used directly by
            // protractor or the browser, but instead pass directly to third parties
            // like SauceLabs as the name of the job running this test
            name: 'Firefox Job',

            // User defined name for the capability that will display in the results log
            // Defaults to the browser name
            logName: 'firefox',

            // Number of times to run this set of capabilities (in parallel, unless
            // limited by maxSessions). Default is 1.
            count: 1,

            // If this is set to be true, specs will be sharded by file (i.e. all
            // files to be run by this set of capabilities will run in parallel).
            // Default is false.
            shardTestFiles: false,

            // Maximum number of browser instances that can run in parallel for this
            // set of capabilities. This is only needed if shardTestFiles is true.
            // Default is 1.
            maxInstances: 2,

            // Additional spec files to be run on this capability only.
            specs: ['*_e2e.js']

        }
    ],

    // Framework to use. Jasmine 2 is recommended.
    framework: 'jasmine2',

    // Options to be passed to Jasmine.
    jasmineNodeOpts: {
        defaultTimeoutInterval: 30000
    }
};
